pkg/volume/csi/csi_attacher.go

```diff
@@ -64,88 +64,124 @@ func (c *csiAttacher) Attach(spec *volume.Spec, nodeName types.NodeName) (string
 		return "", errors.New("missing spec")
 	}
 
-	csiSource, err := getCSISourceFromSpec(spec)
+	var (
+		driverName   string
+		volumeHandle string
+		pvSource     *v1.CSIPersistentVolumeSource
+		node         = string(nodeName)
+	)
+	volSource, pvSource, err := getSourceFromSpec(spec)
 	if err != nil {
 		klog.Error(log("attacher.Attach failed to get CSI persistent source: %v", err))
 		return "", err
 	}
 
-	skip, err := c.plugin.skipAttach(csiSource.Driver)
+	if c.plugin.inlineFeatureEnabled && volSource != nil {
+		driverName = volSource.Driver
+		if volSource.VolumeHandle != nil {
+			volumeHandle = *volSource.VolumeHandle
```

Comment:
why is this one a pointer that needs a nil check whereas the `volumeHandle` in `pvSource` is just a string